src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClassInnerAndNestedClass.java

```diff
@@ -0,0 +1,76 @@
+/*
+FinalClass
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.design.finalclass;
+
+public class InputFinalClassInnerAndNestedClass {
+
+    private class SuperClass { // ok
+        private SuperClass() {
+        }
+    }
+
+    private class SubClass extends SuperClass {
+    }
```

Comment:
This looks like exact test case from https://github.com/checkstyle/checkstyle/issues/4037#issue-215182468, so this PR will also close #4037

@pbludov, if you agree, let's make a comment in #10859 so that we do not forget to close #4037

Reply:
done.

Reply:
Resolved